src/main/java/org/stellar/base/AssetTypeNative.java

```diff
@@ -7,6 +7,16 @@
   public AssetTypeNative() {}
 
   @Override
+  public String getCode() {
```

Comment:
I guess but my point is you don't really need those functions. what you really want is `public boolean equals(Object object)` You might as well implement that separately in each class derived from Asset.